src/BuildScriptGeneratorCli/Commands/BuildCommandBaseBinder.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Binding;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli.Commands
+{
+    public abstract class BuildCommandBaseBinder<T> : CommandBaseBinder<T>
+        where T : BuildCommandBaseProperty
+    {
+#pragma warning disable SA1401 // Fields should be private
```

Comment:
Nit: Please consider adding to the comment why we don't make these fields private, so that future devs reading this code don't need to dig too much to understand why.

Reply:
I'll definitely add more docs and comments to explain it for maintainability.